sympy/printing/pretty/tests/test_pretty.py

```diff
@@ -3199,6 +3199,18 @@ def test_pretty_ConditionSet():
     assert pretty(ConditionSet(x, Eq(sin(x), 0), S.Reals)) == ascii_str
     assert upretty(ConditionSet(x, Eq(sin(x), 0), S.Reals)) == ucode_str
 
+    assert pretty(ConditionSet(x, Contains(x, S.Reals, evaluate=False), FiniteSet(1))) == \
+        '{x | x in {1} and (Contains(x, (-oo, oo)))}'
+    assert upretty(ConditionSet(x, Contains(x, S.Reals, evaluate=False), FiniteSet(1))) == u('{x | x ∊ {1} ∧ (x ∈ ℝ)}')
+
+    assert pretty(ConditionSet(x, And(x > 1, x < -1), FiniteSet(1, 2, 3))) ==\
+        '{x | x in {1, 2, 3} and (And(x > 1, x < -1))}'
+    assert upretty(ConditionSet(x, And(x > 1, x < -1), FiniteSet(1, 2, 3))) == \
+        u('{x | x ∊ {1, 2, 3} ∧ (x > 1 ∧ x < -1)}')
+
+    assert pretty(ConditionSet(x, Or(x > 1, x < -1), FiniteSet(1, 2))) == '{x | x in {1, 2} and (Or(x > 1, x < -1))}'
+    assert upretty(ConditionSet(x, Or(x > 1, x < -1), FiniteSet(1, 2))) == u('{x | x ∊ {1, 2} ∧ (x > 1 ∨ x < -1)}')
```

Comment:
but here, yes

A test for Interval should be added, too. 

But we should maybe stop and think about this. The second arg to ConditionSet is supposed to be a condition for x and the third argument the interval from which x is selected. None of the examples you have added make sense in light of this. (And all of them, it seems, should have been evaluated, e.g. `ConditionSet(x, Or(x > 1, x < -1), FiniteSet(1,2))` should just give `FiniteSet(2)`. That is a separate issue, however.)

Let's go back and see if it is the instantiation that should have raised an error for bad args -- in which case the printer would not need to be modified: we would just raise an error at instantiation.